pkg/controller/snapshot_controller.go

```diff
@@ -421,13 +421,65 @@ func (ctrl *csiSnapshotController) checkandBindSnapshotContent(snapshot *crdv1.V
 	return nil
 }
 
+func (ctrl *csiSnapshotController) prepareCreateSnapshotInput(snapshot *crdv1.VolumeSnapshot) (*crdv1.VolumeSnapshotClass, *v1.PersistentVolume, string, map[string]string, error) {
```

Comment:
Since this function is extract from the `createSnapshotOperation`  function, so we can call this function in the `createSnapshotOperation` to reduce the repeat code